frontend/server/handlers/artifacts.ts

```diff
@@ -74,6 +76,96 @@ export interface GCSProviderInfo {
   };
 }
 
+/**
+ * Returns an authorization middleware for artifact endpoints.
+ * This middleware handles 3 modes:
+ *
+ * 1. Standalone KFP deployment without Kubeflow platform (single-tenant):
+ *    No namespace parameter, no Subject Access Review, 100% insecure.
+ *
+ * 2. Default multi-tenant deployment of KFP within Kubeflow platform:
+ *    Namespace parameter provided, validate format and check RBAC (the user
+ *    is authenticated to access the artifact from the specific namespace
+ *    folder on the object storage via Subject Access Review) and access
+ *    SeaweedFS/storage directly.
+ *
+ * 3. Artifact PROXY MODE (overhead, disabled by default):
```

Comment:
The per namespace artifact proxy mount the credentials per namespace and proxies requests.